netbout-web/src/main/java/com/netbout/dynamo/DyMessages.java

```diff
@@ -202,7 +204,14 @@ public long unread() throws IOException {
             )
             .where(DyFriends.HASH, Conditions.equalTo(this.bout))
             .where(DyFriends.RANGE, Conditions.equalTo(this.self))
-            .iterator().next();
+            .iterator();
+        if (!iterator.hasNext()) {
+            throw new RsFailure(
```

Comment:
@original-brownbear there is no way to delete a bout.. but if user has no access to the bout it looks like missed bout in this method

I agree with @2686747 there is a cache issue caused by an unsafe implementation on `DyMessages.unread()` 

can we just return 0 if no record found ? if the user has no access to bout, so he has 0 unread messages, right ?

> @dmzaytsev care to answer this ? To me it looks like this situation should simply be handled like a missing bout, but maybe you have more to say on this ? :)